src/LoadingError.php

```diff
@@ -0,0 +1,147 @@
+<?php
+/**
+ * Class LoadingError.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP;
+
+use AmpProject\AmpWP\Infrastructure\Service;
+
+/**
+ * Client-side app loading error markup and styles.
+ *
+ * @package AmpProject\AmpWP
+ * @since 2.1.3
+ * @internal
+ */
+final class LoadingError implements Service {
+
+	/**
+	 * Render error message along with necessary styles.
+	 */
+	public function render() {
+		?>
+		<div id="amp-pre-loading-spinner" class="amp-spinner-container">
+			<span class="amp-loading-spinner">
+				<span class="screen-reader-text">
+					<?php esc_html_e( 'Loading', 'amp' ); ?>
+				</span>
+			</span>
+		</div>
+
+		<div id="amp-loading-failure" class="error-screen-container">
+			<div class="error-screen components-panel">
+				<h1>
+					<?php esc_html_e( 'Something went wrong.', 'amp' ); ?>
+				</h1>
+
+				<?php
+				printf(
+					'<p class="amp-loading-failure-script">%s</p>',
+					wp_kses(
+						sprintf(
+							/* translators: %s is the AMP support forum URL. */
+							__( 'Check your connection and open your browser console to see if there are any error messages. You may share them on the <a href="%s" target="_blank" rel="noreferrer noopener">support forum</a>.', 'amp' ),
+							esc_url( __( 'https://wordpress.org/support/plugin/amp/', 'amp' ) )
+						),
+						[
+							'a' => [
+								'href'   => true,
+								'target' => true,
+								'rel'    => true,
+							],
+						]
+					)
+				);
+				?>
+
+				<noscript>
+					<p class="amp-loading-failure-noscript"><?php esc_html_e( 'You must have JavaScript enabled to use this page.', 'amp' ); ?></p>
+				</noscript>
+			</div>
+		</div>
+		<style>
+			#amp-loading-failure {
+				visibility: hidden;
+				animation: amp-wp-show-element 30s steps(1, end) 0s 1 normal both;
+			}
+
+			#amp-pre-loading-spinner {
+				visibility: visible;
+				animation: amp-wp-hide-element 30s steps(1, end) 0s 1 normal both; /* This could probably reuse amp-wp-show-element if reversed. */
+			}
+
+			.amp-loading-failure-noscript {
+				display: none;
+			}
+
+			@keyframes amp-wp-show-element {
+				from {
+					visibility: hidden;
+				}
+				to {
+					visibility: visible;
+				}
+			}
+
+			@keyframes amp-wp-hide-element {
+				from {
+					visibility: visible;
+				}
+				to {
+					visibility: hidden;
+				}
+			}
+
+			@keyframes amp-loading-spinner {
+				from {
+					transform: rotate(0deg);
+				}
+				to {
+					transform: rotate(360deg);
+				}
+			}
+
+			body.no-js #amp-loading-failure {
+				animation: none;
+				visibility: visible;
+			}
+
+			body.no-js .amp-loading-failure-noscript {
+				display: block;
+			}
+
+			body.no-js #amp-pre-loading-spinner,
+			body.no-js .amp-loading-failure-script {
+				display: none;
+			}
+
+			.amp-spinner-container .amp-loading-spinner {
+				display: inline-block;
+				background-color: #949494;
+				width: 18px;
+				height: 18px;
+				opacity: 0.7;
+				margin: 5px 11px 0;
```

Comment:
@delawski why the `margin`? This is not present in the `Loading` component and that's causing the slight layout shift when the static spinner is replaced with the React one. I think this line simply should be removed:

```suggestion
```

Reply:
Seems this was copied from `spinner-stylers.js`:

![image](https://user-images.githubusercontent.com/134745/123167953-29cecb80-d42c-11eb-905c-790b643dadbc.png)

Buy as you can see the `margin` is overridden by `assets/src/components/loading/style.css`:

https://github.com/ampproject/amp-wp/blob/a958b4356074983d589e997bb518922aea8dd76f/assets/src/components/loading/style.css#L7-L9

So yeah, I'm just going to remove the `margin` to make it consistent.